### PR TITLE
Update delay in docs

### LINKED ISSUE #512: Correct documentation on reaction delay
Currently the dart docs for `reaction` state:
```
You can also pass in an optional [name], a debouncing [delay] in milliseconds.
``` 
As noted in #101 this delay is actually throttle, not a debounce. To understand the difference, play with this http://demo.nimius.net/debounce_throttle/

I suggest
```
You can also pass in an optional [name], a throttling [delay] in milliseconds.
```

### PR DESCRIPTION
**Pull request by MisterJimson**
Fixes #512

### CODE REVIEW

File: mobx/lib/src/api/reaction.dart [hunk 1]
```diff
@@ -6,7 +6,7 @@ import 'package:mobx/src/core.dart';
 ///
 /// Optional configuration:
 /// * [name]: debug name for this reaction
-/// * [delay]: debouncing delay in milliseconds
+/// * [delay]: throttling delay in milliseconds
```
Comment (pavanpodila) [inline mobx/lib/src/api/reaction.dart:9]: There are other reactions like autorun that have the same kind of docs. Would be good to sweep through other reaction APIs and replace debouncing with throttling

Comment (MisterJimson) [reply mobx/lib/src/api/reaction.dart:9]: I updated autorun, I will search for others and report back.

**Comment by MisterJimson** [comment]
I updated the only other place I could find "debounce" in the repo.

**Comment by pavanpodila** [comment]
@allContributors add @MisterJimson for doc

**Pull request merged**